kernel/src/hil/storage_interface.rs

```diff
@@ -0,0 +1,74 @@
+//! Interfaces for storage devices.
```

Comment:
Perhaps rename this file `log.rs`?

Reply:
I believe that there is a plan to implement a couple other storage interfaces in the future. Is it alright to continue calling this `storage_interface.rs` under the assumption that they will be added to this file some time in the future, or should this file be renamed and the new storage interfaces go into their own files?

Reply:
If they share elements (like counter/alarm/timer) then a single file makes sense (although it should still be called "storage.rs". If they don't, then it seems easier to just have them be separate.

We actually have examples both ways on this (i2c/spi/uart are not grouped, but ninedof/humidity/light/temp are). I'm not sure which is better, but from a git point of view it is usually easier to have different files.